### ISSUE TITLE
Silent server disconnections

### ISSUE THREAD

**Issue by Twi5TeD** [ORelio/Minecraft-Console-Client#581]
so recently my console client has been loading then after couple mins the alts disconnect without no message and the console screen just shows the chat. does anyone know how to fix this?

i have couple things setup

kickmessages 
```
Connection has been lost
Server is restarting
Server is full
too many people
disconnect.spam
You are not whitelisted on this server!
Server closed
Server restarting
The server you were previously on went down, you have been connected to a fallback server
You seem to be logged in already.
Internal Exception: java.io.IOException: An existing connection was forcibly closed by the remote host.
It may help to try logging in again in a few minutes.
If this does not resolve your issue, please contact staff.
Not connected to any server. Use '/help' for help.
You are already logged on to this server.
[AutoRelog] Waiting 100 seconds before reconnecting...
```

matches.ini
```
[Match]
match= to PvP Wars!
actionother=send /server fire
```

minecraftclient.ini
```
[Main]
login=
password=
serverip=
language=en_GB
consoletitle=%username%@%serverip% - Minecraft Console Client
internalcmdchar=slash
splitmessagedelay=2
botowners=Player1,Player2,Player3 
botmessagedelay=2
mcversion=1.8.0
brandinfo=mcc
chatbotlogfile=
privatemsgscmdname=msg
showsystemmessages=true
showxpbarmessages=false
showchatlinks=false
terrainandmovements=true
sessioncache=disk
resolvesrvrecords=fast
accountlist=accounts.txt
serverlist=servers.txt
playerheadicon=false
exitonfailure=false
debugmessages=false
scriptcache=false
timestamps=false

[MCSettings]
enabled=true
locale=en_US
renderdistance=tiny
difficulty=peaceful
chatmode=enabled
chatcolors=false
main_hand=left
skin_cape=false
skin_hat=false
skin_jacket=false
skin_sleeve_left=false
skin_sleeve_right=false
skin_pants_left=false
skin_pants_right=false

[AntiAFK]
enabled=true
delay=600
command=/server fire

[AutoRelog]
enabled=true
delay=100
retries=-1
kickmessagesfile=kickmessages.txt

[ScriptScheduler]
enabled=true
tasksfile=tasks.ini

[AutoRespond]
enabled=true
matchesfile=matches.ini
```

rejoin.txt
`send /server fire`

tasks.ini
```
[Task]
triggerOnFirstLogin=true
triggerOnLogin=true
triggerOnInterval=true
timeInterval=60
script=rejoin.txt
```

**Comment by ORelio** [comment]
If using Mac/Linux, the Mono framework may cause silent disconnections, I never found out why.

**Comment by Twi5TeD** [comment]
im using windows

**Comment by ORelio** [comment]
Ok, if the disconnection is silent, then I'm not sure why it happens. As a workaround, you may try to use AntiAFK with a useless command such as /ping, let's say, every few minutes. It should help detecting that connection is lost since sending the command will fail. That should trigger the AutoRelog.

**Comment by CarbonNeuron** [comment]
I've had a lot of experience with the silent disconnects. The only real solution I had was to do Application.restart() every 12 hours. This may seem like a short amount of time, but it really isnt. The antiAFK bot also didn't help in my case. It just filled the console with: "sending (command)" even throughout the silent disconnects. I've also figured out if you have the client doing a command that returns text, for example /f who which returns information about a faction, it will keep the client alive longer. 

Makes me wonder if garbage collection is throwing away the connection objects.

**Comment by xthrane** [comment]
> I've had a lot of experience with the silent disconnects. The only real solution I had was to do Application.restart() every 12 hours. This may seem like a short amount of time, but it really isnt. The antiAFK bot also didn't help in my case. It just filled the console with: "sending (command)" even throughout the silent disconnects. I've also figured out if you have the client doing a command that returns text, for example /f who which returns information about a faction, it will keep the client alive longer.
> 
> Makes me wonder if garbage collection is throwing away the connection objects.

How did you make that Application restart? I'm completely new to scripts and stuff, so an elaboration would be wonderful :)

**Comment by CarbonNeuron** [comment]
I basically just modified the antiAFK bot to every 12 hours call Application.restart()

**Comment by Jccse** [comment]
Same thing is happening to me when I put my alts up, after like 2-30 minutes they disconnect for no reason , tried everything I can think of.. I give up..

**Comment by ORelio** [comment]
Silent disconnections should be resolved as per #802. Please try out the latest [development build](https://ci.appveyor.com/project/ORelio/minecraft-console-client/build/artifacts).

**Comment by JackMakesCode** [comment]
I'm getting this issue, gonna try a few of those things. Will put here if it works
